### ISSUE TITLE
Traefik sometimes return 400 Bad Request

### ISSUE THREAD

**Issue by ubajze** [traefik/traefik#10379]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I have Traefik deployed in Kubernetes to forward requests to backend services. Services are exposed via tcp/8443 using TLS. I can successfully get responses in most cases, but it happens from time to time that Traefik returns the 400 Bad Request in plain HTTP.

The Python HTTP client throws an error similar to this one:
```
HTTPSConnectionPool(host='myhost.dev', port=443): Max retries exceeded with url: /api/resource/ (Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1002)')))
```

I am using the `TLSOptions` configuration for services. The following is the K8s manifest that I use:
```
apiVersion: traefik.io/v1alpha1
kind: TLSOption
metadata:
  labels:
    app.kubernetes.io/instance: traefik-private
    endpoint: private
    kustomize.toolkit.fluxcd.io/name: traefik-private
  name: ssl-options-private
  namespace: traefik
spec:
  cipherSuites:
  - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
  - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
  - TLS_AES_256_GCM_SHA384
  - TLS_CHACHA20_POLY1305_SHA256
  curvePreferences:
  - CurveP521
  - CurveP384
  minVersion: VersionTLS12
  sniStrict: true
```

I have an AWS network load balancer between clients and Traefik. So the actual path is `client` -> `NLB:443` -> `Traefik:8443`.

To isolate the problem, I analyzed traffic with Wireshark. I could see the "bad request" message coming from Traefik pods in response to the Client Hello in TLS. This is how the packet looks in the Wireshark:
```
Frame 5924: 56 bytes on wire (448 bits), 56 bytes captured (448 bits) on interface utun4, id 0
Null/Loopback
Internet Protocol Version 4, Src: 10.6.1.201, Dst: 172.16.0.2
Transmission Control Protocol, Src Port: 443, Dst Port: 57578, Seq: 104, Ack: 518, Len: 0
[2 Reassembled TCP Segments (103 bytes): #5922(103), #5924(0)]
Hypertext Transfer Protocol
    [Expert Info (Warning/Security): Unencrypted HTTP protocol detected over encrypted port, could indicate a dangerous misconfiguration.]
    HTTP/1.1 400 Bad Request\r\n
        [Expert Info (Chat/Sequence): HTTP/1.1 400 Bad Request\r\n]
        Response Version: HTTP/1.1
        Status Code: 400
        [Status Code Description: Bad Request]
        Response Phrase: Bad Request
    Content-Type: text/plain; charset=utf-8\r\n
    Connection: close\r\n
    \r\n
    [HTTP response 1/1]
    File Data: 15 bytes
Line-based text data: text/plain (1 lines)
```
 

### What did you see instead?

I see a `400 Bad Request` returned in plain HTTP.

### What version of Traefik are you using?

v2.10.6

### What is your environment & configuration?

Ingress:

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt
    traefik.ingress.kubernetes.io/router.entrypoints: websecure
    traefik.ingress.kubernetes.io/router.middlewares: traefik-hsts-private@kubernetescrd
    traefik.ingress.kubernetes.io/router.tls.options: traefik-ssl-options-private@kubernetescrd
  name: myservice
spec:
  ingressClassName: traefik
  rules:
  - host: myhost.dev
    http:
      paths:
      - backend:
          service:
            name: myservice
            port:
              name: http
        path: /
        pathType: Prefix
  tls:
  - hosts:
    - myhost.dev
    secretName: certificate
```

Middleware:
```yaml
spec:
  headers:
    browserXssFilter: true
    contentTypeNosniff: true
    frameDeny: false
    sslRedirect: true
    stsIncludeSubdomains: true
    stsPreload: true
    stsSeconds: 31536000
```

TLSOptions:
```yaml
apiVersion: traefik.io/v1alpha1
kind: TLSOption
metadata:
  name: ssl-options-private
spec:
  cipherSuites:
  - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
  - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
  - TLS_AES_256_GCM_SHA384
  - TLS_CHACHA20_POLY1305_SHA256
  curvePreferences:
  - CurveP521
  - CurveP384
  minVersion: VersionTLS12
  sniStrict: true
```

**Comment by nmengin** [comment]
Hey @ubajze ,
Thanks for reaching out.

Could you provide a minimal reproducible case (for instance, full Kubernetes manifest to reproduce the issue)?

In the meantime time, if any community member can help us find verified steps to reproduce and fix the issue if possible, we would love the help.

**Comment by ubajze** [comment]
@nmengin Sorry for the late response. I think this was the issue: https://github.com/traefik/traefik/pull/10320

We no longer saw those errors when we upgraded Traefik to `v2.11.0-rc1`.

Just so you know...
- When I ran a test script very close to Traefik (the same VPC in AWS), I rarely saw the issue, and it was hard to replicate it.
- When I ran a test script a bit further (outside AWS), and the delay was about ~200ms, I saw errors, but I had to run my test script for a while.
- When I ran a test script even further, where the delay was about ~600ms, the reproduction rate was super high. The issue was seen in less than 50 requests for sure.

Let's close the issue for now.
